compose/material3/material3/src/jsWasmMain/kotlin/androidx/compose/material3/PlatformDateFormat.jsWasm.kt

```diff
@@ -0,0 +1,207 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.material3
+
+import androidx.compose.ui.text.intl.Locale
+import kotlin.js.Date
+import kotlinx.datetime.Instant
+import kotlinx.datetime.LocalDate
+import kotlinx.datetime.TimeZone
+import kotlinx.datetime.atStartOfDayIn
+import kotlinx.datetime.toLocalDateTime
+
+internal actual object PlatformDateFormat {
+
+    actual val weekdayNames: List<Pair<String, String>>?
+        get() = weekdayNames()
+
+    actual val firstDayOfWeek: Int
+        get() = firstDayOfWeek()
+
+    private const val NUMERIC = "numeric"
+    private const val SHORT = "short"
+    private const val LONG = "long"
+
+    private val countriesFirstDow: Map<String, Int> by lazy {
+        listOf(
+            7 to listOf("TH", "ET", "SG", "SG", "JM", "BT", "IN", "US", "US", "MO", "KE", "IN", "DO", "AU", "IL", "IN", "KE", "KE", "AS", "TW", "IN", "MZ", "MM", "CN", "IN", "PR", "IN", "PK", "BD", "NP", "IN", "KE", "HN", "BR", "HK", "KE", "PK", "CN", "IN", "TT", "ZA", "VE", "KE", "US", "IN", "IN", "MT", "BD", "ZA", "ET", "ET", "MO", "PH", "PE", "ZA", "ID", "DM", "IL", "CN", "WS", "ZW", "TH", "MO", "PK", "PK", "UM", "LA", "ZA", "BZ", "IN", "MM", "PK", "SG", "KE", "IN", "JP", "SV", "BR", "IN", "IN", "CN", "KE", "BD", "PK", "CN", "KE", "BR", "CN", "US", "KE", "KE", "IN", "HK", "IN", "KE", "US", "ET", "CN", "IN", "IN", "SA", "PH", "ET", "MT", "IL", "IN", "US", "PH", "PH", "CO", "GT", "ZW", "KE", "MO", "JP", "BZ", "US", "IN", "BW", "IL", "ZW", "KE", "SG", "ET", "LA", "JP", "KE", "KR", "ID", "IN", "HK", "PE", "MZ", "HK", "IN", "PA", "IN", "KE", "MZ", "YE", "BS", "KE", "IN", "KE", "IN", "KR", "MX", "ZA", "IN", "IN", "BD", "ZA", "IN", "MH", "ID", "ZA", "KE", "IN", "KE", "MT", "IN", "MZ", "ID", "CN", "PK", "IN", "PH", "GU", "PY", "IN", "BT", "PH", "MO", "US", "AG", "KE", "PE", "ID", "KE", "US", "CA", "KE", "ZW", "KE", "ZW", "SG", "KH", "PR", "CN", "IN", "KE", "IN", "KE", "HK", "NP", "IN", "ID", "IN", "ID", "TW", "IN", "HK", "KE", "IN", "TH", "PT", "VI", "KH", "IN", "IN", "IN", "MZ", "CA", "US", "IN", "ET", "IN", "NI", "CN", "PK", "IN"),
+            6 to listOf("EG", "AF", "SY", "AF", "IR", "OM", "IQ", "DZ", "IR", "DJ", "DJ", "IR", "AE", "IR", "SD", "AF", "IQ", "KW", "JO", "DZ", "AF", "IQ", "SD", "IR", "DZ", "DZ", "AF", "IR", "AE", "BH", "QA", "EG", "IQ", "IR", "LY", "SY", "DJ")
+        ).map { (day, tags) -> tags.map { it to day } }.flatten().toMap()
+    }
+
+    //TODO: replace formatting with kotlinx datetime when supported
```

Comment:
Is there a ticket/feature request for kotlinx datetime?

Reply:
Kotlinx datetime [will soon support formatting and parsing](https://github.com/Kotlin/kotlinx-datetime/pull/251#issuecomment-1655603790)